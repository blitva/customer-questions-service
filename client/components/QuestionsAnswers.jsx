import React, { useState } from 'react';
import styled from 'styled-components';

const QandAContainer = styled.div`
  // width: 735px;
  height: 126px;
  padding-left: 7.25px;
  display: block;
  float: left;
  margin-left: 40px;
`;

const QuestionBlock = styled.div`
  display: flex;
  flex-direction: row;
  margin-top: 15px;
  margin-bottom: 8px;
`;

const AnswersBlock = styled(QuestionBlock)`
  margin-top: 8px;
  margin-bottom: 12px;
`;

const QuestionHeader = styled.div`font-weight: 700;`;
const AnswerHeader = styled(QuestionHeader)``;

const Question = styled.div`
  margin-left: 30px;
`;

const Answers = styled(Question)`
  width: 625px;
  margin-left: 40px;
  margin-bottom: 8px;
  color: #0F1111
`;

const User = styled.span`
  color: #565959!important;
`;

const SeeMoreAnswers = styled.div`
  width: 625px;
  height: 20px;
  padding-top: 6px;
`;

const Links = styled.a`
  color: #007185;
`;

const QuestionsAnswers = () => {
  return (
    <QandAContainer>
      <QuestionBlock>
        <QuestionHeader>Question:</QuestionHeader>
        <Question>
          <Links href="#">
            Can it be paired with another identical Echo for true stereo?
          </Links>
        </Question>
      </QuestionBlock>
      <AnswersBlock>
        <AnswerHeader>Answer:</AnswerHeader>
        <Answers>
          Yes. I just set up a system with a show or dot in every room for my elderly mother less than a week ago. You can either create a group of just the couple/few that you want to work together, or if you want the music to play on all the echos, dots, and shows simultaneously just say "Alexa, play (name your music) everywhere." "Everywhere" is the key command, and that works without defining or setting up a group. <br/>
          <User>By Janet W. on October 22, 2020</User><br/>
          <SeeMoreAnswers>
            <Links href="#">
              See more answers (39)
            </Links>
          </SeeMoreAnswers>
        </Answers>
      </AnswersBlock>
    </QandAContainer>
  )
}

export default QuestionsAnswers;