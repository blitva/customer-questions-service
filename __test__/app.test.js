import React from 'react';
import { shallow, mount, configure } from 'enzyme';
import sinon from 'sinon';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });
import CustomerQuestions from '../client/components/App.jsx';
import QuestionsAnswers from '../client/components/QuestionsAnswers.jsx';

describe('<CustomerQuestions />', () => {
  it('renders without crashing', () => {
    shallow(<CustomerQuestions />);
  });

  it('renders correctly', () => {
    const wrapper = shallow(<CustomerQuestions />);
    expect(wrapper).toMatchSnapshot();
  });

  // it('calls useEffect hook', () => {
  //   sinon.spy(CustomerQuestions.prototype, 'useEffect');
  //   const wrapper = mount(<CustomerQuestions />);
  //   expect(CustomerQuestions.prototype.useEffect).to.have.property('callCount', 1);
  // });

  // it('allows us to set props', () => {
  //   const wrapper = mount(<CustomerQuestions question="Do pigs fly?" />);
  //   expect(wrapper.props().question).to.equal('Do pigs fly?');
  //   wrapper.setProps({ questions: 'Do pigs fly?'});
  //   expect(wrapper.props().bar).to.equal('Do pigs fly?');
  // })

  // it('passes props to <Votes />', () => {
  //   const wrapper = mount(<CustomerQuestion />);
  //   const
  // })
});