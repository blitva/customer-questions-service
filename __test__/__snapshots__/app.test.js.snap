// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CustomerQuestions /> renders correctly 1`] = `ShallowWrapper {}`;
